crates/dids/src/methods/dht/document_packet/mod.rs

```diff
@@ -1,26 +1,439 @@
+                        .to_string(),
+                ))?;
+            let vm = VerificationMethod::from_resource_record(&did_uri, record, *idx == 0)?;
+            idx_to_vm_id.insert(*idx, vm.id.clone());
+            verification_methods.push(vm);
+        }
+
+        // 2. Reconstitute verification relationships
+        // 2.1 Reconstitute assertion methods
+        let assertion_method = reconstitute_verification_relationship(
+            &root_record.asm,
+            &idx_to_vm_id,
+            "assertion method",
+        )?;
+
+        // 2.2 Reconstitute authentication methods
+        let authentication = reconstitute_verification_relationship(
+            &root_record.auth,
+            &idx_to_vm_id,
+            "authentication method",
+        )?;
+
+        // 2.3 Reconstitute key agreement
+        let key_agreement = reconstitute_verification_relationship(
+            &root_record.agm,
+            &idx_to_vm_id,
+            "key agreement",
+        )?;
+
+        // 2.4 Reconstitute capability invocations
+        let capability_invocation = reconstitute_verification_relationship(
+            &root_record.inv,
+            &idx_to_vm_id,
+            "capability invocation",
+        )?;
+
+        // 2.5 Reconstitute capability delegations
+        let capability_delegation = reconstitute_verification_relationship(
+            &root_record.del,
+            &idx_to_vm_id,
+            "capability delegation",
+        )?;
+        // 3. Reconstitute services
+        let mut services: Vec<Service> = vec![];
+        for idx in root_record.srv {
+            let record = answers
+                .iter()
+                .find(|record| Service::is_service_record_with_index(record, idx))
+                .ok_or(DocumentPacketError::RootRecord(
+                    "Root record contains reference to service record that is not found"
+                        .to_string(),
+                ))?;
+
+            let service = Service::from_resource_record(&did_uri, record)?;
+            services.push(service);
+        }
+
+        let service = if services.is_empty() {
+            None
+        } else {
+            Some(services)
+        };
+
+        // 4. Reconstitute controllers
+        let controller = answers
+            .iter()
+            .find(|record| Controller::is_cnt_record(record));
+        let controller: Option<Vec<String>> = match controller {
+            None => None,
+            Some(cnt) => Some(Controller::from_resource_record(cnt)?),
+        };
+
+        // 5. Reconstitute alsoKnownAs
+        let also_known_as = answers
+            .iter()
+            .find(|record| AlsoKnownAs::is_aka_record(record));
+        let also_known_as = match also_known_as {
+            None => None,
+            Some(aka) => Some(AlsoKnownAs::from_resource_record(aka)?),
+        };
+
+        // 6. Create document
+        Ok(Document {
+            id: did_uri,
+            context: None,
```

Comment:
Is context: None on purpose?

Reply:
Yep, it is. `did:dht` does not use `context`

 https://did-dht.com/#note-5
<img width="814" alt="Screenshot 2024-06-03 at 12 36 47 PM" src="https://github.com/TBD54566975/web5-rs/assets/74206001/53f65c8a-e4e8-4dc6-aa9a-fbd75a2370cd">=